### PR TITLE
Update to wgpu 23.0.0

### PR DESCRIPTION
**Pull request by waywardmonkeys**
This also updates the MSRV to 1.76 to match the updated MSRV of wgpu 23.

### CODE REVIEW

File: vello/src/wgpu_engine.rs [hunk 1]
```diff
@@ -827,7 +827,7 @@ impl WgpuEngine {
             label: Some(label),
             layout: Some(&compute_pipeline_layout),
             module: &shader_module,
-            entry_point: "main",
+            entry_point: Some("main"),
```
Comment (DJMcNab) [inline vello/src/wgpu_engine.rs:830]: I'd like to try:
```suggestion
            entry_point: None,
```
here

**Review by DJMcNab — APPROVED** [review]
Barring wgpu-profiler, this looks correct.

Nice that our MSRV checks are working as designed

File: vello/src/wgpu_engine.rs (hunk 1 above)
Comment (waywardmonkeys) [reply vello/src/wgpu_engine.rs]: Done.

**Comment by waywardmonkeys** [comment]
@DJMcNab This is now failing on macOS only in the `many_bins_gpu` test that you added as part of #680 / #681. Confirmed that it happens locally on this branch and not on current `main`.

Thoughts or pointers?

**Comment by DJMcNab** [comment]
That is extraordinarily bizarre; it seems to have fixed #680, but only on macOS? That doesn't make sense.

I don't really know how to begin debugging this, as I don't have a Mac.
One thing to try is reproducing #680 in an actual window - i.e make a window larger than your screen using the "drag down, expand/resize up" trick.

That test is effectively a reverse regression test, i.e. that if we fix the mentioned issue, we should also close #680.
I'd be reasonably happy to just ignore the test on macOS and move on...

**Review by DJMcNab — APPROVED** [review]

**Pull request merged**